### PR TITLE
API package cleanup

### PR DESCRIPTION
**Pull request by adriacidre**
Small cleanup of api package

Important changes:
- [x] Avoid exposing outside the package unused functions / types
- [x] Coverage slightly improved
- [x] Improved documentation

Closes #

### CODE REVIEW

File: api/backend_test.go [hunk 1]
```diff
@@ -245,6 +257,32 @@ func TestBackendCallRPCConcurrently(t *testing.T) {
...
+	backend := NewStatusBackend()
+
+	var testCases = []struct {
+		name          string
+		fromState     appState
+		toState       appState
+		expectedState appState
+	}{
+		{
+			name:          "lol",
```
Comment (adambabik) [inline api/backend_test.go]: `lol`? :D And just one test case? Maybe `AppStateChange` should be also verified with concurrent requests? 

Overall, `parseAppState()` used in `AppStateChange` should have a unit test with test cases. `AppStateChange()` should only verify that `backend` is properly updated.

**Comment by adambabik** [comment]
Overall LGTM 👍

File: api/backend_test.go (hunk 1 above)
Comment (adriacidre) [reply api/backend_test.go]: LOL!

About concurrent requests I can implement them, but, it seems `AppStateChange` is waiting for a final implementation. I've added an internal StatusBackend state to help with the tests, but, other than that, it just logs the state change:

```go
// AppStateChange handles app state changes (background/foreground).
// state values: see https://facebook.github.io/react-native/docs/appstate.html
func (b *StatusBackend) AppStateChange(state string) {
	s, err := parseAppState(state)
	if err != nil {
		log.Error("AppStateChange failed, ignoring", "error", err)
		return // and do nothing
	}

	b.log.Info("App State changed", "new-state", s)
	b.appState = s

	// TODO: put node in low-power mode if the app is in background (or inactive)
	// and normal mode if the app is in foreground.
}
```
I think makes sense to wait for the final implementation before going deeper with testing.

**Review by adambabik — APPROVED** [review]
LGTM

**Review by gravityblast — APPROVED** [review]
LGTM

**Pull request merged**